prost-build/src/code_generator.rs

```diff
@@ -1145,15 +1169,20 @@ fn build_enum_value_mappings<'a>(
 ) -> Vec<EnumVariantMapping<'a>> {
     let mut numbers = HashSet::new();
     let mut generated_names = HashMap::new();
-    let mut mappings = Vec::new();
+    let mut mappings: Vec<EnumVariantMapping<'a>> = Vec::new();
 
     for (idx, value) in enum_values.iter().enumerate() {
-        // Skip duplicate enum values. Protobuf allows this when the
+        // Remember name for duplicate enum values. Protobuf allows this when the
         // 'allow_alias' option is set.
+        let mut alias_of = None;
         if !numbers.insert(value.number()) {
-            continue;
+            for m in &mappings {
+                if m.proto_number == value.number() {
+                    alias_of = Some(m.generated_variant_name.clone());
```

Comment:
So the first value is the original and any values with the same integer value are aliases. Do I understand correct?

Reply:
Yes, in this rust representation that's the case, I don't think protobuf itself makes any distinction there, it simply allows the same value to be repeated.